rules/combo/dropper/shell.yara

```diff
@@ -43,12 +43,24 @@ rule sleep_rm : notable {
     any of them
 }
 
-
 rule nohup_bash_background : suspicious {
   strings:
 	$ref = /nohup bash [\%\w\/\>]{0,64} &/
   condition:
     any of them
 }
 
+rule fetch_pipe_shell_value : suspicious {
+  meta:
+	description = "fetches content and pipes it to a shell"
+  strings:
+	$wget_bash = /wget .{8,128}\| {0,2}bash/
```

Comment:
Just to make sure that I understand what this is saying. Any character after wget as long as it's between 8-128 in length? Why are we capping it at 128? And I guess conversely, why starting at 8? This wouldn't catch something like:
wget foo.io

Or am I misunderstanding something?

Also, I can't quite grok what the `{0,2}` is matching.

Reply:
For performance reasons, YARA rules discourage unbounded regular expressions: https://github.com/Neo23x0/YARA-Performance-Guidelines/

But yeah, this says that any wget command line greater than 128 characters won't match. Otherwise, the wget and bash could potentially be 1GB apart.

I started at 8 because I figured you would have to specify the protocol, but to pipe to a shell,l you'll still need to provide `-O- foo.io` at a minimum, which is still 10  characters. 

the {0,2} will match 0 to 2 spaces.

Reply:
```
export F='-O- example.com'
wget $F
```

Would it catch that?

Also, for curl below it's going to stdout by default, and don't need a protocol.

Reply:
Also, since generally the url limit is 2048, maybe bump to that?

Reply:
No, it won't catch the `export` case or other obfuscations. 

To keep the scope small, the philosophy I've been going with for bincapz is to catch behaviors we've seen in the wild, and that's one I haven't seen yet. I have seen people rename curl though!

If I go up to 2048 for a URL, it would increase false-positive rates as the "." will include random binary content. I have not seen any malware samples where folks embed overly large URLs - it's more trouble than it's worth. 128 is already more than twice what has been observed.

Reply:
Ok, I'm just curious how much we want to try to be proactive vs. reactive here. Seems like renaming files might be another file that would catch these. So with the combination of rules, I'd expect things to be more tightened up. But, I trust you here :)